src/lib/NumberIncrementer/index.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { FunctionComponent, useState, useCallback } from 'react'
+import { createCounterSubtractFactory, createCounterAddFactory } from './factory.counter'
+import clsx from 'clsx'
+
+import styles from './numberIncrementer.module.scss'
+
+interface Props {
+  className?: string
+  [rest: string]: unknown // ...rest property
+}
+
+const NumberIncrementer: FunctionComponent<Props> = ({ className, ...rest }) => {
+  const [number, setNumber] = useState(1)
+
+  const subtractNumber = useCallback(() => {
+    setNumber(createCounterSubtractFactory(number))
+  }, [number])
+
+  const addNumber = useCallback(() => {
+    setNumber(createCounterAddFactory(number))
+  }, [number])
+
+  return (
+    <div className={clsx(styles['number-incrementer-wrapper'], className)} {...rest}>
+      <button onClick={subtractNumber} data-testid="subtract" aria-label="subtract number">
```

Comment:
instead of adding a `data-testid` for testing, you can query this element by `aria-label` with `getByLabelText`. https://testing-library.com/docs/react-testing-library/cheatsheet#queries